src/main/java/pages/yandex/HomePage.java

```diff
@@ -14,7 +14,6 @@
     public HomePage(WebDriver driver) {
```

Comment:
Do not pass WebDriver instance as parameter. You have singleton class to access it from where you want